src/mono/mono/metadata/w32handle.c

```diff
@@ -508,7 +508,7 @@ mono_w32handle_ops_prewait (MonoW32Handle *handle_data)
 static void
 mono_w32handle_unlock_handles (MonoW32Handle **handles_data, gsize nhandles)
 {
-	for (gsize i = nhandles - 1; i >= 0; i--) {
+	for (int i = ((int)nhandles - 1); i >= 0; i--) {
 		if (!handles_data [i])
 			continue;
 		mono_w32handle_unlock (handles_data [i]);
```

Comment:
Or we can keep gsize here like this.
```suggestion
	for (gsize i = nhandles; i > 0; i--) {
		if (!handles_data [i - 1])
			continue;
		mono_w32handle_unlock (handles_data [i - 1]);
```

Reply:
Just restored old behavior, you could argue if it should be reverse at all, also discovered that this function is not even called, so occurs to be dead code. Will remove complete instance in a different PR, but lets this one stay for now so tests can run and we can unblock the issues the other fixes in this PR should resolve.